app/models/UserAnswers.scala

```diff
@@ -31,9 +31,12 @@ final case class UserAnswers(
   def get[A](page: Gettable[A])(implicit rds: Reads[A]): Option[A] =
     Reads.optionNoError(Reads.at(page.path)).reads(data).getOrElse(None)
 
-  def set[A](page: Settable[A], value: A)(implicit writes: Writes[A]): Try[UserAnswers] = {
+  def set[A](page: Settable[A] with Gettable[A], newValue: A)(implicit writes: Writes[A], rds: Reads[A]): Try[UserAnswers] = {
 
-    val updatedData = data.setObject(page.path, Json.toJson(value)) match {
+    val oldValue   = get(page)
+    val hasChanged = !oldValue.contains(newValue)
```

Comment:
I can see why you've done it like this to be extra clear about what it's doing but I do think something like `val hasValueChanged = !get(page).contains(value)` would be fine

Reply:
I'm fine with that change. Done